drivers/ksfilter/ks/irp.c

```diff
@@ -370,18 +370,31 @@ KsQueryInformationFile(
             /* use FastIoQueryBasicInfo routine */
             if (FastIoDispatch->FastIoQueryBasicInfo)
             {
-                return FastIoDispatch->FastIoQueryBasicInfo(FileObject, TRUE, (PFILE_BASIC_INFORMATION)FileInformation, &IoStatus, DeviceObject);
+                Result = FastIoDispatch->FastIoQueryBasicInfo(FileObject, TRUE, (PFILE_BASIC_INFORMATION)FileInformation, &IoStatusBlock, DeviceObject);
```

Comment:
I'd prefer combine conditions, but generally it's fine
```
if (FastIoDispatch->FastIoQueryBasicInfo != NULL && FastIoDispatch->FastIoQueryBasicInfo(...))
{
return IoStatusBlock.Status;
}
```

Reply:
Done, though `!= NULL` seems superfluous.